src/utils/beacon/SignPayloadRequestModal.tsx

```diff
@@ -59,14 +63,16 @@ export const SignPayloadRequestModal: React.FC<{
           {`${request.appMetadata.name}/dApp Pairing Request`}
         </Heading>
         <Box
+          overflowY="auto"
+          maxHeight="300px"
           padding="15px"
           border="1px solid"
           borderColor={colors.gray[500]}
           borderRadius="4px"
           backgroundColor={colors.gray[800]}
         >
           <Text color={colors.gray[450]} size="md">
-            {request.payload}
+            {payloadDecoder(request.payload)}
```

Comment:
it might be that we cannot parse the payload (it's just an arbitrary text converted to a hex string, you can pass something like 0501000000000000, which cannot be converted to a string)
so, in such case, the function should throw an exception, and here we should display the actual payload